### PR TITLE
Add sync experiment dependencies

### PR DESCRIPTION
**Pull request by hyperrealist**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR updates `meta.yaml` using suggestions from `grayskull pypi nslsii`. Also the build number is incremented, and the list of recipe maintainers updated.

### CODE REVIEW

**Comment by hyperrealist** [comment]
@mrakitin the CI failures seem to be related to curio. I ran into this before and wrote an [issue](https://github.com/dabeaz/curio/issues/367). As far as I know they have fixed it in the git repo, but they are not publishing to pypi anymore.

**Comment by mrakitin** [comment]
If it's only the Windows builds failing, we can just exclude them. We don't use this package for non-Linux installations.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,79 +6,63 @@ package:
...
-    - nslsii.devices
-    # The following error happened for Win/Py312:
-    #   AttributeError: module 'multiprocessing.connection' has no attribute 'CHALLENGE'. Did you mean: '_CHALLENGE'?
-    # Excluding the test for that platform/Python version for now.
-    - nslsii.iocs.eps_two_state_ioc_sim  # [not (win and py==312)]
-    # Uncomment the next line once https://github.com/NSLS-II/nslsii/issues/129
-    # is resolved.
-    # - nslsii.iocs.thermo_sim
-    - nslsii.temperature_controllers
-    - nslsii.transforms
```
Comment (mrakitin) [inline recipe/meta.yaml:70]: Some imports are still useful to capture potentially missing dependencies as `__init__` on the `nslsii` level may not have everything exposed. Can we revert those? I have a command which can generate the list: https://gist.github.com/mrakitin/b53d00d7ea167f3c7515f2929dd1199c.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -6,79 +6,63 @@ package:
...
 build:
-  number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
-  skip: true  # [py<37 or osx]
-  ignore_run_exports:
-    - python_abi
+  entry_points:
+    - sync-experiment = nslsii.sync_experiment:main
+    - what-is-ioc = nslsii.epics_utils:main
+  noarch: python
```
Comment (mrakitin) [inline recipe/meta.yaml:16]: If we decide to skip Windows builds, this line needs to be replaced with `skip: true  # [win]` and then the feedstock should be rerendered.

**Review by mrakitin — COMMENTED** [review]
Left my review comments. Thanks for the contribution, @hyperrealist!

**Pull request closed without merging**